### DISCUSSION TITLE
Is there a debug mode for OBS?

### DISCUSSION THREAD

**Discussion by ghost** [obsproject/obs-studio#8973] [Development Questions] [answered]
Hi,

I wanted to know if there was a debug mode or if one was already in the works. This has been an issue for me and I made the very bad decision to make my own debug console that attached itself to the Main Application and would return all the function calls into a console interface and I could then dump it into a file.

Thanks,
ɅLΞX | YouTube

**Comment by WizardCM** [comment, marked as answer]
Yes, OBS has a Debug mode. You want to build in the Debug configuration (`CMAKE_BUILD_TYPE`) and attach a Debugger.

On Windows, this is done with Visual Studio. On macOS, this is done with Xcode, and on Linux this can be done with something like [Builder](https://wiki.gnome.org/Apps/Builder) with the Flatpak.

Then you just breakpoint the code path you're interested in and step through. If you have follow-up questions you can ask in the #development channel in the [OBS Community Discord](https://obsproject.com/discord) server.
